[CI] GenerativeIT test failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #7835 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/eloopw3taucm4)
- [elasticsearch-pull-request #69189 / part-3](https://gradle-enterprise.elastic.co/s/ebrhcqlk2fsue)
- [elasticsearch-pull-request #69186 / part-3](https://gradle-enterprise.elastic.co/s/xmuvf7u6vjzos)
- [elasticsearch-pull-request #69156 / part-3](https://gradle-enterprise.elastic.co/s/fbkuxcpg4epvm)
- [elasticsearch-pull-request #69087 / part-3](https://gradle-enterprise.elastic.co/s/7h6jhmm3wcdzc)
- [elasticsearch-pull-request #69079 / part-3](https://gradle-enterprise.elastic.co/s/dzz3tkyyuakzi)
- [elasticsearch-periodic-platform-support #7829 / ubuntu-2404-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/bogvdyte5f7w6)
- [elasticsearch-periodic #7789 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/cbqszlfdc6sk4)
- [elasticsearch-periodic-platform-support #7811 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/6borir7rac5fi)
- [elasticsearch-pull-request #68588 / part-3](https://gradle-enterprise.elastic.co/s/l26y65ksoq2te)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT.test" -Dtests.seed=61E6C94C946D0B3A -Dtests.locale=es-US -Dtests.timezone=Greenwich -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: from languages_lookup,message_types | dissect type "%{language_name} %{lUExNPlCFv}" | keep `language_name`, `message`, `type`, `language_code` | drop messag*, language_name | where  NOT false AND true | stats  dZHmcwpytj = min(language_code), language_code = count(*) by type | limit 3173 | enrich languages_policy on type | keep language_code, dZHmcwpytj| rename dZHmcwpytj as language_code | lookup join languages_lookup on language_code | dissect language_name "%{rzEYPygMrvRn} %{VoxwLDJkW}" | dissect rzEYPygMrvRn "%{rzEYPygMrvRn} %{oRDpAlpSPJrS}" | eval  `oRDpAlpSPJrS` = 4069359665708631012, NRIVvYrpf = null, eusQAcRMXvpY = null, pKCqMKIRHa = null, `rzEYPygMrvRn` = true, lnZPOkBpIVRS = -342457484, rzEYPygMrvRn = 1213716962 | dissect VoxwLDJkW "%{CqJzWvklnYU} %{kpENLsKnqNC}" | rename `rzEYPygMrvRn` AS `sdxbyBdM`, `language_name` AS CqJzWvklnYU, NRIVvYrpf AS `mqMgkAlp` | keep `eusQAcRMXvpY`, CqJzWvklnYU, `oRDpAlpSPJrS`, sdxbyBdM, eusQAcRMXvpY, `kpENLsKnqNC
[truncated]
```

**Issue Reasons:**
- [main] 2 consecutive failures in test test
- [main] 11 failures in test test (1.2% fail rate in 881 executions)
- [main] 7 failures in step part-3 (2.3% fail rate in 298 executions)
- [main] 3 failures in pipeline elasticsearch-periodic-platform-support (16.7% fail rate in 18 executions)
- [main] 7 failures in pipeline elasticsearch-pull-request (2.4% fail rate in 295 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -429,9 +429,6 @@ tests:
 - class: org.elasticsearch.search.query.QueryPhaseTimeoutTests
   method: testBulkScorerTimeout
   issue: https://github.com/elastic/elasticsearch/issues/127156
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/127157
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test026InstallBundledRepositoryPluginsViaConfigFile
   issue: https://github.com/elastic/elasticsearch/issues/127158

--- a/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/GenerativeIT.java
+++ b/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/GenerativeIT.java
@@ -15,7 +15,7 @@ import org.elasticsearch.xpack.esql.qa.rest.generative.GenerativeRestTest;
 import org.junit.ClassRule;
 
 /**
- * This test generates random queries, runs them agains the CSV test dataset and checks that they don't throw unexpected exceptions.
+ * This test generates random queries, runs them against the CSV test dataset and checks that they don't throw unexpected exceptions.
  *
  * If muted, please:
  * <ul>

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
@@ -37,7 +37,7 @@ public abstract class GenerativeRestTest extends ESRestTestCase {
         "Reference \\[.*\\] is ambiguous",
         "Cannot use field \\[.*\\] due to ambiguities",
         "cannot sort on .*",
-        "argument of \\[count_distinct\\(.*\\)\\] must",
+        "argument of \\[count.*\\] must",
         "Cannot use field \\[.*\\] with unsupported type \\[.*_range\\]",
         "Unbounded sort not supported yet",
         "The field names are too complex to process", // field_caps problem
@@ -51,6 +51,7 @@ public abstract class GenerativeRestTest extends ESRestTestCase {
         "Unknown column \\[<all-fields-projected>\\]", // https://github.com/elastic/elasticsearch/issues/121741,
         "Plan \\[ProjectExec\\[\\[<no-fields>.* optimized incorrectly due to missing references", // https://github.com/elastic/elasticsearch/issues/125866
         "token recognition error at: '``", // https://github.com/elastic/elasticsearch/issues/125870
+                                           // https://github.com/elastic/elasticsearch/issues/127167
         "Unknown column \\[.*\\]", // https://github.com/elastic/elasticsearch/issues/126026
         "optimized incorrectly due to missing references", // https://github.com/elastic/elasticsearch/issues/116781
         "No matches found for pattern", // https://github.com/elastic/elasticsearch/issues/126418

